smartsnippets_inherit/static/admin/smartsnippet_inherit/js/plugininherit_form.js

```diff
@@ -0,0 +1,205 @@
+                    }
+                });
+            };
+        },
+        resetToInitialVariables: function(snippetId){
+            return function(){ _editor.loadVariables(snippetId) };
+        },
+        loadVariables: function (snippetId) {
+            var self = _editor;
+            SnippetWidgetRegistry.deregisterAllVariables();
+            // show variables
+            var varsBox = $(self.options.variables);
+            $(varsBox).html(self.getInfoBox("Loading variables...", "progress"));
+            $.ajax({
+                url: self.options.urls.variables,
+                data: {'snippet_plugin': snippetId},
+                success: function (data) {
+                    $(varsBox).html(data).append(
+                        $('<button type="button" />')
+                            .text('Reset variables to original values')
+                            .click(self.resetToOriginalVariables(snippetId)),
+
+                        $('<button type="button" />')
+                            .text('Reset changed variables')
+                            .click(self.resetToInitialVariables(snippetId)),
+
+                        $('<button type="button" />')
+                            .text('Overwrite changed variables')
+                            .click(self.submitOverwriteData(snippetId))
+                    );
+                    SnippetWidgetRegistry.initializeVariables();
+                    self.registerInitialData(snippetId);
+                },
+                error: function(){
+                    $(varsBox).html(self.getInfoBox("Error loading variables", "error"));
+                },
+                complete: function(){
+                    self.resizeIframe();
+                }
+            });
+        },
+        initSnippetPlugins: function(){
+            var self = this;
+            $(self.options.snippetItems).each(function(){
+                $(this).click(function(){
+                    // mark selected
+                    $(this).addClass('selected').siblings().removeClass('selected');
+                    var snippetId = parseInt($(this).attr('data-snippet'), 10);
+                    self.loadVariables(snippetId);
+                });
+            });
+            $('form').submit(function () {
+                return !self._variablesSubmission;
+            });
+        },
+        initOpts: function(opts){
+            this.options = $.extend(opts, {
+                snippets: opts.variables_module + ' .snippets-list',
+                snippetItems: opts.variables_module + ' .snippets-list li',
+                variables: opts.variables_module + ' .variables-list'
+            });
+        },
+        setHeaders: function () {
+
+            function csrfSafeMethod(method) {
+                // these HTTP methods do not require CSRF protection
+                return (/^(GET|HEAD|OPTIONS|TRACE)$/.test(method));
+            }
+
+            $.ajaxSetup({
```

Comment:
This will change the behavior all future ajax calls which is not recommended (as per decumentation http://api.jquery.com/jquery.ajaxsetup/). It will also override all previously set ajaxsetup on django.jquery.
I would recommend 2 options:
1. subclass jQuery : var myNewJquery = $.sub();
//and if you don't want to rewrite all references to $ in you code : var $ = $.sub();
2. append ajax setup to each of you ajax calls:
//define you ajax setup
var ajaxSetup = {
    global: true,
    beforeSend: function(xhr, settings) {
                   //...
                },
    cache: false
}

$.ajax($.extend( {
   url: 'someurl',
   success: function(){}
  //...
}, ajaxSetup));